lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -848,7 +854,9 @@ sub _restart ($self, %args) {
     my @urls;
     for (my $i = 0; $i < @$duplicates; $i++) {
         my $result = $duplicates->[$i];
-        $self->emit_event(openqa_job_restart => {id => $jobs->[$i], result => $result, auto => $auto});
+        my %event_data = (id => $jobs->[$i], result => $result, auto => $auto);
+        $event_data{comment} = $comment if defined $comment;
+        $self->emit_event(openqa_job_restart => \%event_data);
```

Comment:
So this would emit the restart event with a comment attribute but no comment creation event would be triggered. Should we instead emit both events?

Reply:
Correct. For the sake of simplicity I just made it one event. However, I can change it to also emit single events for comments. (Will be a bit more code changes to pass around all the information.)